src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3991,19 +3982,10 @@ private static bool CanReuseParameter(CSharp.Syntax.ParameterSyntax parameter)
                     name = AddTrailingSkippedSyntax(name, SyntaxList.List(open, close));
                 }
             }
-            else if (this.IsPossibleName())
```

Comment:
Similar issue, we only hit this branch if hasArgList is true.  And if hasArgList is true, then IsPossibleName returns false as it expects only a "true identifier" (which means the syntax kind must actually be SyntaxKind.Identifier, and not SyntaxKind.ArgListKeyword).